application: serial_lte_modem: Data mode enhancement (#8758)
nrfconnect/sdk-nrf: applications/serial_lte_modem/doc/slm_data_mode.rst

The SLM application sends an *OK* response when it successfully enters data mode.

Sending data in data mode
=========================

If the current sending function succeeds and :ref:`CONFIG_SLM_DATAMODE_URC <CONFIG_SLM_DATAMODE_URC>` is defined, the SLM application reports back a total size as ``#XDATAMODE: <size>``. The ``<size>`` should be a value of positive integer.

Exiting data mode
=================

To exit data mode, the MCU sends the termination command set by the :ref:`CONFIG_SLM_DATAMODE_TERMINATOR <CONFIG_SLM_DATAMODE_TERMINATOR>` configuration option over UART.

The pattern string could be sent alone or as an affix to the data.

> If the current sending function fails, the SLM application exits data mode and returns an error as ``#XDATAMODE: <error>``. The ``<error>`` should be a value of negative integer.

Reviewer: ```suggestion
If the current sending function fails, the SLM application exits data mode and returns an error as ``#XDATAMODE: <error>``.
The ``<error>`` value is a negative integer.
```

   This option specifies a pattern string to terminate data mode.
   The default pattern string is ``+++``.

.. _CONFIG_SLM_DATAMODE_URC:

CONFIG_SLM_DATAMODE_URC - Send URC in data mode

>    This option reports result of previous data sending, while the SLM application stays in data mode.

Reviewer: ```suggestion
   This option reports the result of the previous data-sending operation while the SLM application remains in data mode.
```

>    MCU could use this URC for application-level uplink flow control.

Reviewer: ```suggestion
   The MCU could use this URC for application-level uplink flow control.
```

::

   #XDATAMODE: <size>
   #XDATAMODE: <error>
   #XDATAMODE: 0

The ``<size>`` value returns an integer indicating the total size of the sending operation in data mode.

> The ``<error>`` value returns an integer indicating the error of the sending operation in data mode.

Reviewer: ```suggestion
The ``<error>`` value returns an integer indicating the error code of the sending operation in data mode.
```

> The ``0`` value indicates that the SLM application quits data modem and returns to AT command mode.

Reviewer: ```suggestion
The ``0`` value indicates that the SLM application quit data mode and returned to AT command mode.
```
